### PR TITLE
Document Stanford's style guide

### PR DESCRIPTION
**Pull request by stvstnfrd**
I've talked about and brainstormed this quite a bit, finally starting to put it into writing.
The goal is to provide a prescriptive guide to the coding conventions used by our team, synthesizing PEP8, Pylint, and our own in-house guidelines.

The overall document will be a long-running work-in-progress, but this represents the first steps toward canonization.

We should merge individual sections, as they are ready.

### CODE REVIEW

**Comment by stvstnfrd** [comment]
@caseylitton As mentioned.

CC: @dcadams @caesar2164 @kluo 
CC: @jlikhuva @jspayd @potsui

https://github.com/stvstnfrd/edx-platform/blob/c6452bdb4f0ff609af03c926fd5f5334f5ffde21/openedx/stanford/STYLE.markdown

**Comment by stvstnfrd** [comment]
Note: Given the amount of code/diff samples, it may be helpful to view this file as rendered by GitHub [1].

[1] https://github.com/stvstnfrd/edx-platform/blob/c6452bdb4f0ff609af03c926fd5f5334f5ffde21/openedx/stanford/STYLE.markdown

**Comment by stvstnfrd** [comment]
Per slack conversation, I'm splitting this out into a multi-file hierarchy.

**Comment by stvstnfrd** [comment]
Barring any hard objections, I'll go ahead and merge this as-is, with additional PRs to (potentially) follow.

File: openedx/stanford/style/python/collections-multiline.markdown [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+```diff
+     'foo',
+-    'bar',
+ ]
+```
+
+```diff
+     'bar',
++     'baz',
```
Comment (caesar2164) [inline openedx/stanford/style/python/collections-multiline.markdown]: extra space!

File: openedx/stanford/style/python/docstrings.markdown [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+### Bad
+```python
+def foo():
+    """ Demonstrate a docstring
+
+    with multiple lines
+    """
+    pass
+```
```
Comment (caesar2164) [inline openedx/stanford/style/python/docstrings.markdown]: maybe also show a "bad" docstring that uses single quotes?

### Bad
```python
def foo():
'''
Demonstrate a docstring
'''
pass
```

Comment (stvstnfrd) [inline openedx/stanford/style/python/docstrings.markdown]: Done above.

Comment (caesar2164) [inline openedx/stanford/style/python/docstrings.markdown]: cool.

**Pull request merged**